modules/core/client/directives/page-title.client.directive.js

```diff
@@ -19,13 +19,17 @@
 
       function listener(event, toState) {
         var applicationCoreTitle = 'MEAN.js',
-          separeteBy = ' - ';
+          separator = ' - ',
+          stateTitle = applicationCoreTitle + separator;
+
+        toState.name.split('.').forEach(function(value, index) {
+          stateTitle = stateTitle + value.charAt(0).toUpperCase() + value.slice(1) + separator;
+        });
         if (toState.data && toState.data.pageTitle) {
-          var stateTitle = $interpolate(toState.data.pageTitle)($state.$current.locals.globals);
-          element.html(applicationCoreTitle + separeteBy + stateTitle);
-        } else {
-          element.html(applicationCoreTitle);
+          stateTitle = $interpolate(stateTitle + toState.data.pageTitle + separator)(($state.$current.locals.globals));
         }
+        stateTitle = stateTitle.slice(0, 0 - separator.length);
+        element.html(stateTitle);
```

Comment:
Agree with you, it's the old code, don't want to change it, but agree. For me, it's important only if you display pageTitle in your content page, in order to add links to every parts. 
Done.